homeassistant/components/trello/config_flow.py

```diff
@@ -0,0 +1,220 @@
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_options = config_entry.options
+        self.trello_adapter = _create_trello_adapter(
+            config_entry.data[CONF_API_KEY], config_entry.data[CONF_API_TOKEN]
+        )
+
+        self.ids_boards: dict[str, dict[str, str]] = {}
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select desired boards to have card counts of per list.
+
+        :param user_input: None initially or the user's selected boards
+        """
+        configured_board_ids = self.config_options[CONF_OPTIONS_BOARDS].keys()
+
+        if user_input is None:
+            self.ids_boards = await self._fetch_all_boards()
+
+            return await self._show_board_form(configured_board_ids)
+
+        await self._remove_deselected_boards(
+            configured_board_ids, user_input[USER_INPUT_BOARD_IDS]
+        )
+
+        user_selected_boards = await self._get_boards_lists(
+            user_input[USER_INPUT_BOARD_IDS]
+        )
+
+        new_config_options = {CONF_OPTIONS_BOARDS: user_selected_boards}
+        return self.async_create_entry(data=new_config_options)
+
+    async def _show_board_form(self, configured_board_ids: list[str]) -> FlowResult:
+        return self.async_show_form(
+            step_id="init",
+            data_schema=_get_board_select_schema(
+                self.ids_boards, list(configured_board_ids)
+            ),
+        )
+
+    async def _remove_deselected_boards(
+        self, configured_board_ids: list[str], user_selected_board_ids: list[str]
+    ) -> None:
+        for configured_board_id in configured_board_ids:
+            if configured_board_id not in user_selected_board_ids:
+                dev_reg = dr.async_get(self.hass)
+                if device := dev_reg.async_get_device(
+                    identifiers={(DOMAIN, configured_board_id)}
+                ):
+                    dev_reg.async_remove_device(device.id)
+
+    async def _fetch_all_boards(self) -> dict[str, dict[str, str]]:
+        return await self.hass.async_add_executor_job(self.trello_adapter.get_boards)
+
+    async def _get_boards_lists(self, board_ids: list[str]) -> dict[str, dict]:
+        return await self.hass.async_add_executor_job(
+            self.trello_adapter.get_board_lists, self.ids_boards, board_ids
+        )
+
+
+def _create_trello_adapter(api_key: str, api_token: str) -> TrelloAdapter:
+    return TrelloAdapter(TrelloClient(api_key=api_key, api_secret=api_token))
+
+
+def _get_board_select_schema(
+    boards: dict[str, dict], default: list[str] | None = None
+) -> Schema:
+    if default is None:
+        default = []
+    options = {key: value["name"] for key, value in boards.items()}
+    return vol.Schema(
+        {vol.Required(USER_INPUT_BOARD_IDS, default=default): cv.multi_select(options)}
+    )
```

Comment:
I am not sure if this should be moved inside both of the flows. I think it would be useful, because then you can use `self.add_suggested_values_to_schema(` to add the defaults

Reply:
I ended up removing the defaulting behavior since it was only needed for the options flow so that won't be needed for now.